src/components/Header.tsx

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { useCallback } from 'react';
+import cn from 'classnames';
+import { Todo } from '../types/Todo';
+import { Error } from '../types/enumError';
+import { addTodo, updateTodo, USER_ID } from '../api/todos';
+
+type Props = {
+  todos: Todo[];
+  activeCount: number;
+  inputRef: React.RefObject<HTMLInputElement>;
+  newTodoTitle: string;
+  setNewTodoTitle: (title: string) => void;
+  isInputDisabled: boolean;
+  setErrorMessage: React.Dispatch<React.SetStateAction<Error>>;
+  setTempTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
+  setIsInputDisabled: React.Dispatch<React.SetStateAction<boolean>>;
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+};
+
+export const Header: React.FC<Props> = ({
+  todos,
+  activeCount,
+  inputRef,
+  newTodoTitle,
+  setNewTodoTitle,
+  isInputDisabled,
+  setErrorMessage,
+  setTempTodo,
+  setIsInputDisabled,
+  setTodos,
+}) => {
+  function addPost(loadingTodo: Todo) {
+    setTempTodo(loadingTodo);
+    setErrorMessage(Error.Default);
+    setIsInputDisabled(true);
+
+    addTodo({
+      userId: loadingTodo.userId,
+      title: loadingTodo.title,
+      completed: loadingTodo.completed,
+    })
+      .then(createdTodo => {
+        setTodos(currentTodos => [...currentTodos, createdTodo]);
+        setNewTodoTitle('');
+      })
+      .catch(() => {
+        setErrorMessage(Error.Add);
+        setTimeout(() => setErrorMessage(Error.Default), 3000);
+      })
+      .finally(() => {
+        setTempTodo(null);
+        setIsInputDisabled(false);
+        inputRef.current?.focus();
+      });
+  }
+
+  function handleSubmit(event: React.FormEvent) {
+    event.preventDefault();
+
+    const trimmedTitle = newTodoTitle.trim();
+
+    if (!trimmedTitle) {
+      setErrorMessage(Error.Empty);
+      setTimeout(() => setErrorMessage(Error.Default), 3000);
+
+      return;
+    }
+
+    const loadingTodo: Todo = {
+      userId: USER_ID,
+      title: trimmedTitle,
+      completed: false,
+      id: 0,
+    };
+
+    setTempTodo(loadingTodo);
+    addPost(loadingTodo);
+  }
+
+  const completeAll = useCallback(async () => {
```

Comment:
```suggestion
  const handleCompleteAll = useCallback(async () => {
```

let's use better naming for functions